Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp

```diff
@@ -0,0 +1,338 @@
+            for (u16 i = 0; i < -byte + 1; ++i) {
+                plane_data[index++] = next_byte;
+            }
+        } else if (byte >= 0) {
+            for (u16 i = 0; i < byte + 1; ++i) {
+                plane_data[index] = (u8)data[read_bytes];
+                read_bytes++;
+                index++;
+            }
+        }
+    }
+
+    return plane_data;
+}
+
+static ErrorOr<void> decode_body_chunk(Chunk body_chunk, ILBMLoadingContext& context)
+{
+    dbgln_if(ILBM_DEBUG, "decode_body_chunk {}", body_chunk.data.size());
+
+    if (context.bm_header.compression == CompressionType::ByteRun) {
+        auto& data = body_chunk.data;
+        // these are the uncompressed interleaved bitmap planes
+        auto plane_data = TRY(uncompress_byte_run(data, context));
+        // that we need to convert to chunky pixel data
+        auto pixel_data = TRY(planar_to_chunky(plane_data, context));
+
+        context.bitmap = TRY(chunky_to_bitmap(context, pixel_data));
+    } else {
+        return Error::from_string_literal("Uncompress body not supported yet");
+    }
+
+    return {};
+}
+
+static ErrorOr<Chunk> decode_iff_chunk_header(ReadonlyBytes chunks)
+{
+    if (chunks.size() < sizeof(ChunkHeader))
+        return Error::from_string_literal("Not enough data for IFF chunk header");
+
+    auto const& header = *bit_cast<ChunkHeader const*>(chunks.data());
+
+    if (chunks.size() < sizeof(ChunkHeader) + header.chunk_size)
+        return Error::from_string_literal("Not enough data for IFF chunk");
+
+    return Chunk { header.chunk_type, { chunks.data() + sizeof(ChunkHeader), header.chunk_size } };
+}
+
+static ErrorOr<Chunk> decode_iff_advance_chunk(ReadonlyBytes& chunks)
+{
+    auto chunk = TRY(decode_iff_chunk_header(chunks));
+
+    chunks = chunks.slice(sizeof(ChunkHeader) + chunk.data.size());
+
+    // add padding if needed
+    if (chunk.data.size() % 2 != 0) {
+        if (chunks.is_empty())
+            return Error::from_string_literal("Missing data for padding byte");
+        if (*chunks.data() != 0)
+            return Error::from_string_literal("Padding byte is not 0");
+        chunks = chunks.slice(1);
+    }
+
+    return chunk;
+}
+
+static ErrorOr<void> decode_iff_chunks(ILBMLoadingContext& context)
+{
+    auto& chunks = context.chunks_cursor;
+
+    dbgln_if(ILBM_DEBUG, "decode_iff_chunks");
+
+    while (!chunks.is_empty()) {
+        auto chunk = TRY(decode_iff_advance_chunk(chunks));
+        if (chunk.type == FourCC("CMAP")) {
+            context.color_table = TRY(decode_cmap_chunk(chunk));
+        } else if (chunk.type == FourCC("BODY")) {
+            TRY(decode_body_chunk(chunk, context));
+            context.state = ILBMLoadingContext::State::BitmapDecoded;
+        } else if (chunk.type == FourCC("CRNG")) {
+            dbgln_if(ILBM_DEBUG, "Chunk:CRNG");
+        }
+    }
+
+    return {};
+}
+
+static ErrorOr<void> decode_ilbm_first_chunk(ILBMLoadingContext& context)
```

Comment:
If the first chunk is named BMHD, why not naming the function `decode_bmhd_chunk` (or something else if you want, just more explicit than the current name)